subt/tools/validator.py

```diff
@@ -86,17 +86,37 @@ def autodetect_name(logfile):
         arr = deserialize(data)
         robot_name = arr[0].decode('ascii')
         return robot_name
+    _, _, data = next(LogReader(logfile))
+
+    data = ast.literal_eval(str(data, 'ascii'))
+    while data:
+        if data[0] != "--robot-name":
+            del data[0]
+            continue
+        return data[1]
     assert False, "Robot name autodetection failed!"
 
 
+def autodetect_pose3d(logfile):
+    streams = lookup_stream_names(logfile)
+    streams = set(s.split('.')[0] for s in streams if s.endswith('.pose3d'))
```

Comment:
this is what I meant that "streams" are not actual original streams - the variable is reused

Reply:
maybe `node` (I know how much you like them ;)) would be better name?

Reply:
oh, right, thanks - improvement in a3a6c10